---
title: "Collapse barcodes to final per-RBD/mutant phenotype scores"
author: "Tyler Starr"
date: "05/14/2022"
output:
  github_document:
    toc: true
    html_preview: false
editor_options: 
  chunk_output_type: inline
---
This notebook reads in the per-barcode sera binding AUCs and collapses barcodes to final mean binding for each variant, and generates some coverage and QC analyses. It also makes some heatmaps and other visualizations of these data.

```{r setup, message=FALSE, warning=FALSE, error=FALSE}
require("knitr")
knitr::opts_chunk$set(echo = T)
knitr::opts_chunk$set(dev.args = list(png = list(type = "cairo")))

#list of packages to install/load
packages = c("yaml","data.table","tidyverse","gridExtra")
#install any packages not already installed
installed_packages <- packages %in% rownames(installed.packages())
if(any(installed_packages == F)){
  install.packages(packages[!installed_packages])
}
#load packages
invisible(lapply(packages, library, character.only=T))

#read in config file
config <- read_yaml("config.yaml")

#make output directory
if(!file.exists(config$final_variant_scores_dir)){
  dir.create(file.path(config$final_variant_scores_dir))
}

#read in file giving spike/RBD indexing concordance between backgrounds
RBD_sites <- read.csv(file=config$RBD_annotation_file,stringsAsFactors=F)
```
Session info for reproducing environment:
```{r print_sessionInfo}
sessionInfo()
```

## Setup

Read in tables of per-barcode sera AUC values, as well as the prior SARSr DMS data with expression effects of mutations.

```{r input_data}
dt <- data.table(read.csv(config$sera_delta_AUC_file),stringsAsFactors=F)[library=="lib40_SARSr-DMS",]

dt_expr <- data.table(read.csv(config$SARSr_lib40_mut_bind_expr),stringsAsFactors=F)[,.(target,wildtype,site,site_SARS2,mutant,mutation,mutation_SARS2,expr,delta_expr,n_bc_expr,expr_rep1,expr_rep2)]
#rename targets to match this repo
dt_expr[target=="PRD0038",target:="PRD-0038"]
dt_expr[target=="SARS-CoV-1",target:="SARS-CoV-1_Urbani"]
dt_expr[target=="SARS-CoV-2",target:="Wuhan_Hu_1"]
```

## Calculate per-variant mean scores

Calculate the mean binding score collapsed by genotype. Also output the number of barcodes across which a variant score was determined in each library.

```{r calculate_mean_scores}
dt[,mean_tetra_AUC:=mean(tetra_AUC,na.rm=T),by=c("library","target","aa_substitutions")]
dt[,sd_tetra_AUC:=sd(tetra_AUC,na.rm=T),by=c("library","target","aa_substitutions")]
dt[,n_bc_tetra_AUC:=sum(!is.na(tetra_AUC)),by=c("library","target","aa_substitutions")]

dt[,mean_tri_AUC:=mean(tri_AUC,na.rm=T),by=c("library","target","aa_substitutions")]
dt[,sd_tri_AUC:=sd(tri_AUC,na.rm=T),by=c("library","target","aa_substitutions")]
dt[,n_bc_tri_AUC:=sum(!is.na(tri_AUC)),by=c("library","target","aa_substitutions")]

dt[,mean_SARS2mono_AUC:=mean(SARS2mono_AUC,na.rm=T),by=c("library","target","aa_substitutions")]
dt[,sd_SARS2mono_AUC:=sd(SARS2mono_AUC,na.rm=T),by=c("library","target","aa_substitutions")]
dt[,n_bc_SARS2mono_AUC:=sum(!is.na(SARS2mono_AUC)),by=c("library","target","aa_substitutions")]

dt[,mean_RmYN02mono_AUC:=mean(RmYN02mono_AUC,na.rm=T),by=c("library","target","aa_substitutions")]
dt[,sd_RmYN02mono_AUC:=sd(RmYN02mono_AUC,na.rm=T),by=c("library","target","aa_substitutions")]
dt[,n_bc_RmYN02mono_AUC:=sum(!is.na(RmYN02mono_AUC)),by=c("library","target","aa_substitutions")]

dt <- unique(dt[,.(library,target,variant_class,aa_substitutions,n_aa_substitutions,
                   mean_tetra_AUC, sd_tetra_AUC, n_bc_tetra_AUC,
                  mean_tri_AUC, sd_tri_AUC, n_bc_tri_AUC,
                  mean_SARS2mono_AUC, sd_SARS2mono_AUC, n_bc_SARS2mono_AUC,
                  mean_RmYN02mono_AUC, sd_RmYN02mono_AUC, n_bc_RmYN02mono_AUC)])
```

Some QC plots. First, look at distribution of number barcodes for single mutant detemrinations. These are 'left-justified' histograms, so the leftmost bar represents the number of genotypes for which no barcodes were collapsed to final measurement in a pool.

```{r hist_n_bc_per_mutant, fig.width=8, fig.height=6, fig.align="center", dpi=300,dev="png"}
par(mfrow=c(2,2))
hist(dt[variant_class=="1 nonsynonymous",n_bc_tetra_AUC],main="tetra sera",right=F,breaks=max(dt[variant_class=="1 nonsynonymous",n_bc_tetra_AUC],na.rm=T),xlab="")
hist(dt[variant_class=="1 nonsynonymous",n_bc_tri_AUC],main="tri sera",right=F,breaks=max(dt[variant_class=="1 nonsynonymous",n_bc_tri_AUC],na.rm=T),xlab="")
hist(dt[variant_class=="1 nonsynonymous",n_bc_SARS2mono_AUC],main="SARS2mono sera",right=F,breaks=max(dt[variant_class=="1 nonsynonymous",n_bc_SARS2mono_AUC],na.rm=T),xlab="")
hist(dt[variant_class=="1 nonsynonymous",n_bc_RmYN02mono_AUC],main="RmYN02mono sera",right=F,breaks=max(dt[variant_class=="1 nonsynonymous",n_bc_RmYN02mono_AUC],na.rm=T),xlab="")

```

What about how SEM tracks with number of barcodes collapsed? This could help for choosing a minimum number of barcodes to use or an SEM cutoff. List fraction above SEM 0.5 on each plot.

```{r sem_v_n-bc, fig.width=8, fig.height=8, fig.align="center", dpi=300,dev="png"}
par(mfrow=c(2,2))
plot(dt[variant_class=="1 nonsynonymous",n_bc_tetra_AUC],
     dt[variant_class=="1 nonsynonymous",sd_tetra_AUC/sqrt(n_bc_tetra_AUC)],
     pch=16,col="#00000005",main="tetra sera",ylab="SEM",xlab="number barcodes collapsed")
abline(h=0.5,col="red",lty=2)
legend("topleft",bty="n",cex=1,legend=paste(format(100*nrow(dt[variant_class=="1 nonsynonymous" & sd_tetra_AUC/sqrt(n_bc_tetra_AUC) > 0.5 & !is.na(mean_tetra_AUC),])/nrow(dt[variant_class=="1 nonsynonymous" & !is.na(mean_tetra_AUC),]),digits=3),"%"))

plot(dt[variant_class=="1 nonsynonymous",n_bc_tri_AUC],
     dt[variant_class=="1 nonsynonymous",sd_tri_AUC/sqrt(n_bc_tri_AUC)],
     pch=16,col="#00000005",main="tri sera",ylab="SEM",xlab="number barcodes collapsed")
abline(h=0.5,col="red",lty=2)
legend("topleft",bty="n",cex=1,legend=paste(format(100*nrow(dt[variant_class=="1 nonsynonymous" & sd_tri_AUC/sqrt(n_bc_tri_AUC) > 0.5 & !is.na(mean_tri_AUC),])/nrow(dt[variant_class=="1 nonsynonymous" & !is.na(mean_tri_AUC),]),digits=3),"%"))

plot(dt[variant_class=="1 nonsynonymous",n_bc_SARS2mono_AUC],
     dt[variant_class=="1 nonsynonymous",sd_SARS2mono_AUC/sqrt(n_bc_SARS2mono_AUC)],
     pch=16,col="#00000005",main="SARS2mono sera",ylab="SEM",xlab="number barcodes collapsed")
abline(h=0.5,col="red",lty=2)
legend("topleft",bty="n",cex=1,legend=paste(format(100*nrow(dt[variant_class=="1 nonsynonymous" & sd_SARS2mono_AUC/sqrt(n_bc_SARS2mono_AUC) > 0.5 & !is.na(mean_SARS2mono_AUC),])/nrow(dt[variant_class=="1 nonsynonymous" & !is.na(mean_SARS2mono_AUC),]),digits=3),"%"))

plot(dt[variant_class=="1 nonsynonymous",n_bc_RmYN02mono_AUC],
     dt[variant_class=="1 nonsynonymous",sd_RmYN02mono_AUC/sqrt(n_bc_RmYN02mono_AUC)],
     pch=16,col="#00000005",main="RmYN02mono sera",ylab="SEM",xlab="number barcodes collapsed")
abline(h=0.5,col="red",lty=2)
legend("topleft",bty="n",cex=1,legend=paste(format(100*nrow(dt[variant_class=="1 nonsynonymous" & sd_RmYN02mono_AUC/sqrt(n_bc_RmYN02mono_AUC) > 0.5 & !is.na(mean_RmYN02mono_AUC),])/nrow(dt[variant_class=="1 nonsynonymous" & !is.na(mean_RmYN02mono_AUC),]),digits=3),"%"))

invisible(dev.print(pdf, paste(config$final_variant_scores_dir,"/lib40_sem_v_n-bc.pdf",sep=""),useDingbats=F))
```

Format into a 'mutation lookup table', where we focus just on the single mutants (and wildtype), breakup the string of mutations, and fill in the table to also include any missing mutants. We also provide indexing of sites according to SARS2 numbering and my alignment and according to "self" numbering

```{r format_mutant_table}
dt_mutant <- dt[variant_class %in% "1 nonsynonymous",]

#split mutation string
#define function to apply
split_mut <- function(x){
  split <- strsplit(x,split="")[[1]]
  return(list(split[1],as.numeric(paste(split[2:(length(split)-1)],collapse="")),split[length(split)]))
}
dt_mutant[,c("wildtype","position","mutant"):=split_mut(as.character(aa_substitutions)),by=aa_substitutions]

dt_mutant <- dt_mutant[,.(library,target,wildtype,position,mutant,
                          mean_tetra_AUC,sd_tetra_AUC,n_bc_tetra_AUC,
                          mean_tri_AUC,sd_tri_AUC,n_bc_tri_AUC,
                          mean_SARS2mono_AUC,sd_SARS2mono_AUC,n_bc_SARS2mono_AUC,
                          mean_RmYN02mono_AUC,sd_RmYN02mono_AUC,n_bc_RmYN02mono_AUC)]

aas <- c("A","C","D","E","F","G","H","I","K","L","M","N","P","Q","R","S","T","V","W","Y")
#fill out missing values in table with a hideous loop, so the table is complete for all mutaitons (including those that are missing). If you are somebody who is reading this code, I apologize.
for(lib in as.character(unique(dt_mutant$lib))){
  for(bg in as.character(unique(dt_mutant$target))){
    for(pos in 1:max(dt_mutant$position)){
      for(aa in aas){
        if(!(aa %in% as.character(dt_mutant[library==lib & target==bg & position==pos,mutant]))){
          dt_mutant <- rbind(dt_mutant,list(lib, bg, dt_mutant[target==bg & position==pos,wildtype][1],pos,aa),fill=T)
        }
      }
    }
  }
}
setkey(dt_mutant,library,target,position,mutant)

#fill in wildtype values -- should vectorize in data table but being so stupid so just going to write for loop
for(lib in as.character(unique(dt_mutant$lib))){
  for(bg in as.character(unique(dt_mutant$target))){
    dt_mutant[library==lib & target==bg & wildtype==mutant, 
              c("mean_tetra_AUC","sd_tetra_AUC","n_bc_tetra_AUC",
                "mean_tri_AUC","sd_tri_AUC","n_bc_tri_AUC",
                "mean_SARS2mono_AUC","sd_SARS2mono_AUC","n_bc_SARS2mono_AUC",
                "mean_RmYN02mono_AUC","sd_RmYN02mono_AUC","n_bc_RmYN02mono_AUC") := 
                dt[library==lib & target==bg & variant_class=="wildtype",
                   .(mean_tetra_AUC,sd_tetra_AUC,n_bc_tetra_AUC,
                     mean_tri_AUC,sd_tri_AUC,n_bc_tri_AUC,
                     mean_SARS2mono_AUC,sd_SARS2mono_AUC,n_bc_SARS2mono_AUC,
                     mean_RmYN02mono_AUC,sd_RmYN02mono_AUC,n_bc_RmYN02mono_AUC)]]
  }
}

#add delta bind measures relative to respective wildtypes
for(lib in as.character(unique(dt_mutant$lib))){
  for(bg in as.character(unique(dt_mutant$target))){
    ref_tetra_AUC <- dt[library==lib & target==bg & variant_class=="wildtype",mean_tetra_AUC]
    dt_mutant[library==lib & target==bg,delta_tetra_AUC := mean_tetra_AUC - ref_tetra_AUC]
    ref_tri_AUC <- dt[library==lib & target==bg & variant_class=="wildtype",mean_tri_AUC]
    dt_mutant[library==lib & target==bg,delta_tri_AUC := mean_tri_AUC - ref_tri_AUC]
    ref_SARS2mono_AUC <- dt[library==lib & target==bg & variant_class=="wildtype",mean_SARS2mono_AUC]
    dt_mutant[library==lib & target==bg,delta_SARS2mono_AUC := mean_SARS2mono_AUC - ref_SARS2mono_AUC]
    ref_RmYN02mono_AUC <- dt[library==lib & target==bg & variant_class=="wildtype",mean_RmYN02mono_AUC]
    dt_mutant[library==lib & target==bg,delta_RmYN02mono_AUC := mean_RmYN02mono_AUC - ref_RmYN02mono_AUC]
  }
}

#reindex sites for each background according to alignment. I will keep two columns: one gives each mutation in the "SARS2" spike indexing, one that gives that spike's indexing

#remove positions past the last index (no wildtype position). Could also fix this above when we fill out this table
dt_mutant <- dt_mutant[!is.na(wildtype)]

#set empty columns to fill with indexed sites
dt_mutant$site <- as.character(NA)
dt_mutant$site_SARS2 <- as.character(NA)

#do by bg with if/else looping
for(i in 1:nrow(dt_mutant)){
  if(dt_mutant[i,target]=="Wuhan_Hu_1"){
    dt_mutant[i,site := as.character(RBD_sites[RBD_sites$site_SARS2_RBD==dt_mutant[i,position] & !is.na(RBD_sites$site_SARS2_RBD),"site_SARS2_spike"])]
    dt_mutant[i,site_SARS2 := as.character(RBD_sites[RBD_sites$site_SARS2_RBD==dt_mutant[i,position] & !is.na(RBD_sites$site_SARS2_RBD),"site_SARS2_spike"])]
  }else if(dt_mutant[i,target]=="RshSTT182"){
    dt_mutant[i,site := as.character(RBD_sites[RBD_sites$site_RshSTT182_RBD==dt_mutant[i,position] & !is.na(RBD_sites$site_RshSTT182_RBD),"site_RshSTT182_spike"])]
    dt_mutant[i,site_SARS2 := as.character(RBD_sites[RBD_sites$site_RshSTT182_RBD==dt_mutant[i,position] & !is.na(RBD_sites$site_RshSTT182_RBD),"site_SARS2_spike"])]
  }else if(dt_mutant[i,target]=="PRD-0038"){
    dt_mutant[i,site:=as.character(RBD_sites[RBD_sites$site_PRD0038_RBD==dt_mutant[i,position] & !is.na(RBD_sites$site_PRD0038_RBD),"site_PRD0038_spike"])]
    dt_mutant[i,site_SARS2 := as.character(RBD_sites[RBD_sites$site_PRD0038_RBD==dt_mutant[i,position] & !is.na(RBD_sites$site_PRD0038_RBD),"site_SARS2_spike"])]
  }else if(dt_mutant[i,target]=="SARS-CoV-1_Urbani"){
    dt_mutant[i,site := as.character(RBD_sites[RBD_sites$site_SARS1_RBD==dt_mutant[i,position] & !is.na(RBD_sites$site_SARS1_RBD),"site_SARS1_spike"])]
    dt_mutant[i,site_SARS2 := as.character(RBD_sites[RBD_sites$site_SARS1_RBD==dt_mutant[i,position] & !is.na(RBD_sites$site_SARS1_RBD),"site_SARS2_spike"])]
  }else if(dt_mutant[i,target]=="RsYN04"){
    dt_mutant[i,site := as.character(RBD_sites[RBD_sites$site_RsYN04_RBD==dt_mutant[i,position] & !is.na(RBD_sites$site_RsYN04_RBD),"site_RsYN04_spike"])]
    dt_mutant[i,site_SARS2 := as.character(RBD_sites[RBD_sites$site_RsYN04_RBD==dt_mutant[i,position] & !is.na(RBD_sites$site_RsYN04_RBD),"site_SARS2_spike"])]
  }else if(dt_mutant[i,target]=="RmYN02"){
    dt_mutant[i,site := as.character(RBD_sites[RBD_sites$site_RmYN02_RBD==dt_mutant[i,position] & !is.na(RBD_sites$site_RmYN02_RBD),"site_RmYN02_spike"])]
    dt_mutant[i,site_SARS2 := as.character(RBD_sites[RBD_sites$site_RmYN02_RBD==dt_mutant[i,position] & !is.na(RBD_sites$site_RmYN02_RBD),"site_SARS2_spike"])]
  }
}

#add single mutation string indicators
dt_mutant[,mutation:=paste(wildtype,site,mutant,sep=""),by=c("wildtype","site","mutant")]
dt_mutant[,mutation_SARS2:=paste(wildtype,site_SARS2,mutant,sep=""),by=c("wildtype","site_SARS2","mutant")]

dt_mutant <- unique(dt_mutant[,.(target,wildtype,site,site_SARS2,mutant,mutation,mutation_SARS2,
                               mean_tetra_AUC,sd_tetra_AUC,delta_tetra_AUC,n_bc_tetra_AUC,
                               mean_tri_AUC,sd_tri_AUC,delta_tri_AUC,n_bc_tri_AUC,
                               mean_SARS2mono_AUC,sd_SARS2mono_AUC,delta_SARS2mono_AUC,n_bc_SARS2mono_AUC,
                               mean_RmYN02mono_AUC,sd_RmYN02mono_AUC,delta_RmYN02mono_AUC,n_bc_RmYN02mono_AUC)])

#rename some of the columns
setnames(dt_mutant,"mean_tetra_AUC","tetra_AUC")
setnames(dt_mutant,"mean_tri_AUC","tri_AUC")
setnames(dt_mutant,"mean_SARS2mono_AUC","SARS2mono_AUC")
setnames(dt_mutant,"mean_RmYN02mono_AUC","RmYN02mono_AUC")

```


Censor any measurements that are from a single bc and/or below a certain SEM? don't do this for now, let's visualize first, do this when collapsing summary metric per site.
```{r censor_n_barcodes_libs, echo=T, fig.width=8, fig.height=4, fig.align="center", dpi=300,dev="png"}
# min_bc <- 2
# min_lib <- 2
#  
# dt_mutant[n_bc_bind < min_bc, c("bind","delta_bind","n_bc_bind","n_libs_bind") := list(NA,NA,NA,NA)]
# dt_mutant[n_bc_expr < min_bc, c("expr","delta_expr","n_bc_expr","n_libs_expr") := list(NA,NA,NA,NA)]
```


Coverage stats on n_barcodes for different measurements in the final pooled measurements.

```{r n_barcode_plots, echo=T, fig.width=8, fig.height=6, fig.align="center", dpi=300,dev="png"}
par(mfrow=c(2,2))
hist(dt_mutant[wildtype!=mutant, n_bc_tetra_AUC],col="gray50",main=paste("mutant tetra_AUC score,\nmedian ",median(dt_mutant[wildtype!=mutant, n_bc_tetra_AUC],na.rm=T),sep=""),right=F,breaks=max(dt_mutant[wildtype!=mutant, n_bc_tetra_AUC],na.rm=T),xlab="")

hist(dt_mutant[wildtype!=mutant, n_bc_tri_AUC],col="gray50",main=paste("mutant tri_AUC score,\nmedian ",median(dt_mutant[wildtype!=mutant, n_bc_tri_AUC],na.rm=T),sep=""),right=F,breaks=max(dt_mutant[wildtype!=mutant, n_bc_tri_AUC],na.rm=T),xlab="")

hist(dt_mutant[wildtype!=mutant, n_bc_SARS2mono_AUC],col="gray50",main=paste("mutant SARS2mono_AUC score,\nmedian ",median(dt_mutant[wildtype!=mutant, n_bc_SARS2mono_AUC],na.rm=T),sep=""),right=F,breaks=max(dt_mutant[wildtype!=mutant, n_bc_SARS2mono_AUC],na.rm=T),xlab="")

hist(dt_mutant[wildtype!=mutant, n_bc_RmYN02mono_AUC],col="gray50",main=paste("mutant RmYN02mono_AUC score,\nmedian ",median(dt_mutant[wildtype!=mutant, n_bc_RmYN02mono_AUC],na.rm=T),sep=""),right=F,breaks=max(dt_mutant[wildtype!=mutant, n_bc_RmYN02mono_AUC],na.rm=T),xlab="")

invisible(dev.print(pdf, paste(config$final_variant_scores_dir,"/lib40_histogram_n_bc_per_geno_pooled-libs.pdf",sep="")))
```
## Expression effects

Integrate the previously measured impacts of mutations on expression, look at applying a correction or a cutoff based on expression?

```{r add_expression, echo=T, fig.width=4, fig.height=4, fig.align="center", dpi=300,dev="png"}
dt_mutant[,c("expr","delta_expr"):=as.numeric(NA)]
for(i in 1:nrow(dt_mutant)){
  bg <- as.character(dt_mutant[i,target]) #have to define these b/c data.table sometimes is weird with in situ referneces when colnames are shared...
  pos <- as.character(dt_mutant[i,site])
  aa <- as.character(dt_mutant[i,mutant])
  dt_mutant[i, c("expr","delta_expr") := dt_expr[as.character(target)==bg & as.character(site)==pos & as.character(mutant)==aa,.(expr,delta_expr)]] 
}
```
Serum binding values really drop off below delta_expr values of ~-1. But even zoomed in on the ~-1, we can see there is a correlation. I wonder if we could do something like a spline or linear model and apply correction?

```{r plot_expression, echo=T, fig.width=6, fig.height=12, fig.align="center", dpi=300,dev="png"}
par(mfrow=c(4,2))
plot(dt_mutant[,delta_expr],dt_mutant[,delta_tetra_AUC],pch=16,col="#00000040",xlab="mutant expression",ylab="mutant sera AUC [tetra]");abline(v=-1,lty=2,col="red",main="tetra")

plot(dt_mutant[delta_expr > -1,delta_expr],dt_mutant[delta_expr > -1,delta_tetra_AUC],pch=16,col="#00000040",xlab="mutant expression",ylab="mutant sera AUC [tetra]"); abline(lm(dt_mutant[delta_expr > -1,delta_tetra_AUC] ~ dt_mutant[delta_expr > -1,delta_expr]),lty=2,col="red")

plot(dt_mutant[,delta_expr],dt_mutant[,delta_tri_AUC],pch=16,col="#00000040",xlab="mutant expression",ylab="mutant sera AUC [tri]");abline(v=-1,lty=2,col="red",main="tri")

plot(dt_mutant[delta_expr > -1,delta_expr],dt_mutant[delta_expr > -1,delta_tri_AUC],pch=16,col="#00000040",xlab="mutant expression",ylab="mutant sera AUC [tri]"); abline(lm(dt_mutant[delta_expr > -1,delta_tri_AUC] ~ dt_mutant[delta_expr > -1,delta_expr]),lty=2,col="red")

plot(dt_mutant[,delta_expr],dt_mutant[,delta_SARS2mono_AUC],pch=16,col="#00000040",xlab="mutant expression",ylab="mutant sera AUC [SARS2mono]");abline(v=-1,lty=2,col="red",main="SARS2mono")

plot(dt_mutant[delta_expr > -1,delta_expr],dt_mutant[delta_expr > -1,delta_SARS2mono_AUC],pch=16,col="#00000040",xlab="mutant expression",ylab="mutant sera AUC [SARS2mono]");  abline(lm(dt_mutant[delta_expr > -1,delta_SARS2mono_AUC] ~ dt_mutant[delta_expr > -1,delta_expr]),lty=2,col="red")

plot(dt_mutant[,delta_expr],dt_mutant[,delta_RmYN02mono_AUC],pch=16,col="#00000040",xlab="mutant expression",ylab="mutant sera AUC [RmYN02mono]");abline(v=-1,lty=2,col="red",main="RmYN02mono")

plot(dt_mutant[delta_expr > -1,delta_expr],dt_mutant[delta_expr > -1,delta_RmYN02mono_AUC],pch=16,col="#00000040",xlab="mutant expression",ylab="mutant sera AUC [RmYN02mono]");  abline(lm(dt_mutant[delta_expr > -1,delta_RmYN02mono_AUC] ~ dt_mutant[delta_expr > -1,delta_expr]),lty=2,col="red")

```

Derive a normalized AUC metric that (i) filters out variants with delta_expr <-1 and (ii) normalizes by the linear trend between delta-AUC and delta-expr in this window

```{r derive_norm_AUC, echo=T, fig.width=8, fig.height=8, fig.align="center", dpi=300,dev="png"}
dt_mutant[,c("delta_tetra_normAUC","delta_tri_normAUC","delta_SARS2mono_normAUC","delta_RmYN02mono_normAUC"):=as.numeric(NA)]

lm_tetra <- lm(dt_mutant[delta_expr > -1,delta_tetra_AUC] ~ dt_mutant[delta_expr > -1,delta_expr])
dt_mutant[delta_expr > -1, delta_tetra_normAUC := delta_tetra_AUC - (lm_tetra$coefficients[2] * delta_expr)]

lm_tri <- lm(dt_mutant[delta_expr > -1,delta_tri_AUC] ~ dt_mutant[delta_expr > -1,delta_expr])
dt_mutant[delta_expr > -1, delta_tri_normAUC := delta_tri_AUC - (lm_tri$coefficients[2] * delta_expr)]

lm_SARS2mono <- lm(dt_mutant[delta_expr > -1,delta_SARS2mono_AUC] ~ dt_mutant[delta_expr > -1,delta_expr])
dt_mutant[delta_expr > -1, delta_SARS2mono_normAUC := delta_SARS2mono_AUC - (lm_SARS2mono$coefficients[2] * delta_expr)]

lm_RmYN02mono <- lm(dt_mutant[delta_expr > -1,delta_RmYN02mono_AUC] ~ dt_mutant[delta_expr > -1,delta_expr])
dt_mutant[delta_expr > -1, delta_RmYN02mono_normAUC := delta_RmYN02mono_AUC - (lm_RmYN02mono$coefficients[2] * delta_expr)]

par(mfrow=c(2,2))

plot(dt_mutant[delta_expr > -1,delta_expr],dt_mutant[delta_expr > -1,delta_tetra_normAUC],pch=16,col="#00000040",xlab="mutant expression",ylab="mutant sera expr-norm AUC [tetra]"); abline(lm(dt_mutant[delta_expr > -1,delta_tetra_normAUC] ~ dt_mutant[delta_expr > -1,delta_expr]),lty=2,col="red")

plot(dt_mutant[delta_expr > -1,delta_expr],dt_mutant[delta_expr > -1,delta_tri_normAUC],pch=16,col="#00000040",xlab="mutant expression",ylab="mutant sera expr-norm AUC [tri]"); abline(lm(dt_mutant[delta_expr > -1,delta_tri_normAUC] ~ dt_mutant[delta_expr > -1,delta_expr]),lty=2,col="red")

plot(dt_mutant[delta_expr > -1,delta_expr],dt_mutant[delta_expr > -1,delta_SARS2mono_normAUC],pch=16,col="#00000040",xlab="mutant expression",ylab="mutant sera expr-norm AUC [SARS2mono]");  abline(lm(dt_mutant[delta_expr > -1,delta_SARS2mono_normAUC] ~ dt_mutant[delta_expr > -1,delta_expr]),lty=2,col="red")

plot(dt_mutant[delta_expr > -1,delta_expr],dt_mutant[delta_expr > -1,delta_RmYN02mono_normAUC],pch=16,col="#00000040",xlab="mutant expression",ylab="mutant sera expr-norm AUC [RmYN02mono]");  abline(lm(dt_mutant[delta_expr > -1,delta_RmYN02mono_normAUC] ~ dt_mutant[delta_expr > -1,delta_expr]),lty=2,col="red")


```
Normalization coefficients were: tetra `r lm_tetra$coefficients[2]`, tri `r lm_tri$coefficients[2]`, SARS2mono `r lm_SARS2mono$coefficients[2]`, and RmYN02mono `r lm_RmYN02mono$coefficients[2]`, for an average coefficient of `r mean(c(lm_tetra$coefficients[2], lm_tri$coefficients[2], lm_SARS2mono$coefficients[2], lm_RmYN02mono$coefficients[2]))`


## Heatmaps!

Make heatmaps of raw AUC as well as delta-AUC *expression-normalized* (and expression-censored).

Order factor variables for plotting.

```{r order_plotting_factors}

#order targets for plotting
dt_mutant[target=="Wuhan_Hu_1",target:="SARS-CoV-2_WH1"]
dt_mutant$target <- factor(dt_mutant$target,levels=c("SARS-CoV-2_WH1","RshSTT182","SARS-CoV-1_Urbani","RsYN04","PRD-0038","RmYN02"))
#order mutant as a factor for grouping by rough biochemical grouping
dt_mutant$mutant <- factor(dt_mutant$mutant, levels=c("C","P","G","V","M","L","I","A","F","W","Y","T","S","N","Q","E","D","H","K","R"))
#order the sites character vector (becuase of 372a number for the PRD0038 insertion)
dt_mutant$site_SARS2 <-factor(dt_mutant$site_SARS2,levels=sort(unique(dt_mutant$site_SARS2)))
dt_mutant$site <-factor(dt_mutant$site,levels=sort(unique(dt_mutant$site)))

#add character vector indicating wildtype to use as plotting symbols for wt
dt_mutant[,wildtype_indicator := ""]
dt_mutant[as.character(mutant)==as.character(wildtype),wildtype_indicator := "x"]

#make temp long-form data frame
temp <- data.table::melt(dt_mutant[, .(target,site,site_SARS2,mutant,wildtype_indicator,
                                      tetra_AUC,delta_tetra_AUC,delta_tetra_normAUC,
                                      tri_AUC,delta_tri_AUC,delta_tri_normAUC,
                                      SARS2mono_AUC,delta_SARS2mono_AUC,delta_SARS2mono_normAUC,
                                      RmYN02mono_AUC,delta_RmYN02mono_AUC,delta_RmYN02mono_normAUC)],
                         id.vars=c("target","site","site_SARS2","mutant","wildtype_indicator"),
                         measure.vars=c("tetra_AUC","delta_tetra_AUC","delta_tetra_normAUC",
                                        "tri_AUC","delta_tri_AUC","delta_tri_normAUC",
                                        "SARS2mono_AUC","delta_SARS2mono_AUC","delta_SARS2mono_normAUC",
                                        "RmYN02mono_AUC","delta_RmYN02mono_AUC","delta_RmYN02mono_normAUC"),
                         variable.name="measurement",value.name="value")

#for method to duplicate aa labels on right side of plot https://github.com/tidyverse/ggplot2/issues/3171
guide_axis_label_trans <- function(label_trans = identity, ...) {
  axis_guide <- guide_axis(...)
  axis_guide$label_trans <- rlang::as_function(label_trans)
  class(axis_guide) <- c("guide_axis_trans", class(axis_guide))
  axis_guide
}

guide_train.guide_axis_trans <- function(x, ...) {
  trained <- NextMethod()
  trained$key$.label <- x$label_trans(trained$key$.label)
  trained
}

```
Make heatmaps faceted by target, showing raw affinity and delta-affinity of muts relative to respective

First, tetra sera

```{r heatmap_DMS_AUC-by-target_tetra-sera, fig.width=25,fig.height=15,fig.align="center", dpi=500,dev="png",echo=T}
p1 <- ggplot(temp[measurement=="tetra_AUC",],aes(site_SARS2,mutant))+geom_tile(aes(fill=value),color="black",lwd=0.1)+
  scale_fill_gradientn(colours=c("#FFFFFF","#003366"),limits=c(0,8.5),na.value="gray70")+
  #scale_fill_gradientn(colours=c("#FFFFFF","#FFFFFF","#003366"),limits=c(5,12),values=c(0,1/7,7/7),na.value="yellow")+ #three notches in case I want to 'censor' closer to the 5 boundary condition
  #scale_x_continuous(expand=c(0,0),breaks=c(331,seq(335,530,by=5)))+
  labs(x="",y="")+theme_classic(base_size=9)+
  coord_equal()+theme(axis.text.x=element_text(angle=90,hjust=1,vjust=0.6,face="bold",size=10),axis.text.y=element_text(face="bold",size=10))+
  facet_wrap(~target,nrow=6)+
  guides(y.sec=guide_axis_label_trans())+
  geom_text(aes(label=wildtype_indicator),size=2,color="gray10")+
  theme(strip.text.x = element_text(size = 18))

p1
invisible(dev.print(pdf, paste(config$final_variant_scores_dir,"/lib40_heatmap_SSM_AUC-by-target_tetra.pdf",sep="")))
```

Same but illustrating delta_AUC by target (each delta noramlized to its wt background), for tetra sera.

```{r heatmap_DMS_delta-AUC-by-target_tetra-sera, fig.width=25,fig.height=15,fig.align="center", dpi=500,dev="png",echo=T}
p1 <- ggplot(temp[measurement=="delta_tetra_AUC",],aes(site_SARS2,mutant))+geom_tile(aes(fill=value),color="black",lwd=0.1)+
  scale_fill_gradientn(colours=c("#A94E35","#A94E35","#F48365","#FFFFFF","#7378B9","#383C6C","#383C6C"),
                       limits=c(-8,2),
                       values=c(0/10,4/10,6/10,8/10,8.5/10,9/10, 10/10),
                       na.value="gray70")+
  #scale_x_continuous(expand=c(0,0),breaks=c(331,seq(335,530,by=5)))+
  labs(x="",y="")+theme_classic(base_size=9)+
  coord_equal()+theme(axis.text.x=element_text(angle=90,hjust=1,vjust=0.6,face="bold",size=10),axis.text.y=element_text(face="bold",size=10))+
  facet_wrap(~target,nrow=6)+
  guides(y.sec=guide_axis_label_trans())+
  geom_text(aes(label=wildtype_indicator),size=2,color="gray10")+
  theme(strip.text.x = element_text(size = 18))

p1
invisible(dev.print(pdf, paste(config$final_variant_scores_dir,"/lib40_heatmap_SSM_delta-AUC-by-target_tetra.pdf",sep="")))
```

Same but illustrating delta_normAUC (expression-normalized+censored) by target (each delta noramlized to its wt background), for tetra sera.

```{r heatmap_DMS_delta-normAUC-by-target_tetra-sera, fig.width=25,fig.height=15,fig.align="center", dpi=500,dev="png",echo=T}
p1 <- ggplot(temp[measurement=="delta_tetra_normAUC",],
             aes(site_SARS2,mutant))+geom_tile(aes(fill=value),color="black",lwd=0.1)+
  scale_fill_gradientn(colours=c("#A94E35","#A94E35","#F48365","#FFFFFF","#7378B9","#383C6C"), #if showing blue color
                       limits=c(-5,2),
                       values=c(0/7,3/7,4/7,5/7,6/7,7/7),
                       na.value="gray70")+
  # scale_fill_gradientn(colours=c("#A94E35","#A94E35","#F48365","#FFFFFF","#FFFFFF","#FFFFFF","#FFFFFF"), #if censoring >0 blue
  #                      limits=c(-5,2),
  #                      values=c(0/7,3/7,4/7,5/7,5.5/7,6/7, 7/7),
  #                      na.value="gray70")+
  #scale_x_continuous(expand=c(0,0),breaks=c(331,seq(335,530,by=5)))+
  labs(x="",y="")+theme_classic(base_size=9)+
  coord_equal()+theme(axis.text.x=element_text(angle=90,hjust=1,vjust=0.6,face="bold",size=10),axis.text.y=element_text(face="bold",size=10))+
  facet_wrap(~target,nrow=6)+
  guides(y.sec=guide_axis_label_trans())+
  geom_text(aes(label=wildtype_indicator),size=2,color="gray10")+
  theme(strip.text.x = element_text(size = 18))

p1
invisible(dev.print(pdf, paste(config$final_variant_scores_dir,"/lib40_heatmap_SSM_delta-normAUC-by-target_tetra.pdf",sep="")))
```

Next, tri sera

```{r heatmap_DMS_AUC-by-target_tri-sera, fig.width=25,fig.height=15,fig.align="center", dpi=500,dev="png",echo=T}
p1 <- ggplot(temp[measurement=="tri_AUC",],aes(site_SARS2,mutant))+geom_tile(aes(fill=value),color="black",lwd=0.1)+
  scale_fill_gradientn(colours=c("#FFFFFF","#003366"),limits=c(0,8.5),na.value="gray70")+
  #scale_fill_gradientn(colours=c("#FFFFFF","#FFFFFF","#003366"),limits=c(5,12),values=c(0,1/7,7/7),na.value="yellow")+ #three notches in case I want to 'censor' closer to the 5 boundary condition
  #scale_x_continuous(expand=c(0,0),breaks=c(331,seq(335,530,by=5)))+
  labs(x="",y="")+theme_classic(base_size=9)+
  coord_equal()+theme(axis.text.x=element_text(angle=90,hjust=1,vjust=0.6,face="bold",size=10),axis.text.y=element_text(face="bold",size=10))+
  facet_wrap(~target,nrow=6)+
  guides(y.sec=guide_axis_label_trans())+
  geom_text(aes(label=wildtype_indicator),size=2,color="gray10")+
  theme(strip.text.x = element_text(size = 18))

p1
invisible(dev.print(pdf, paste(config$final_variant_scores_dir,"/lib40_heatmap_SSM_AUC-by-target_tri.pdf",sep="")))
```

Same but illustrating delta_AUC by target (each delta noramlized to its wt background), for tri sera.

```{r heatmap_DMS_delta-AUC-by-target_tri-sera, fig.width=25,fig.height=15,fig.align="center", dpi=500,dev="png",echo=T}
p1 <- ggplot(temp[measurement=="delta_tri_AUC",],aes(site_SARS2,mutant))+geom_tile(aes(fill=value),color="black",lwd=0.1)+
  scale_fill_gradientn(colours=c("#A94E35","#A94E35","#F48365","#FFFFFF","#7378B9","#383C6C","#383C6C"),
                       limits=c(-8,2),
                       values=c(0/10,4/10,6/10,8/10,8.5/10,9/10, 10/10),
                       na.value="gray70")+
  #scale_x_continuous(expand=c(0,0),breaks=c(331,seq(335,530,by=5)))+
  labs(x="",y="")+theme_classic(base_size=9)+
  coord_equal()+theme(axis.text.x=element_text(angle=90,hjust=1,vjust=0.6,face="bold",size=10),axis.text.y=element_text(face="bold",size=10))+
  facet_wrap(~target,nrow=6)+
  guides(y.sec=guide_axis_label_trans())+
  geom_text(aes(label=wildtype_indicator),size=2,color="gray10")+
  theme(strip.text.x = element_text(size = 18))

p1
invisible(dev.print(pdf, paste(config$final_variant_scores_dir,"/lib40_heatmap_SSM_delta-AUC-by-target_tri.pdf",sep="")))
```

expression-normalized/censored, tri

```{r heatmap_DMS_delta-normAUC-by-target_tri-sera, fig.width=25,fig.height=15,fig.align="center", dpi=500,dev="png",echo=T}
p1 <- ggplot(temp[measurement=="delta_tri_normAUC",],
             aes(site_SARS2,mutant))+geom_tile(aes(fill=value),color="black",lwd=0.1)+
  scale_fill_gradientn(colours=c("#A94E35","#A94E35","#F48365","#FFFFFF","#7378B9","#383C6C"), #if showing blue color
                       limits=c(-5,2),
                       values=c(0/7,3/7,4/7,5/7,6/7,7/7),
                       na.value="gray70")+
  # scale_fill_gradientn(colours=c("#A94E35","#A94E35","#F48365","#FFFFFF","#FFFFFF","#FFFFFF","#FFFFFF"), #if censoring >0 blue
  #                      limits=c(-5,2),
  #                      values=c(0/7,3/7,4/7,5/7,5.5/7,6/7, 7/7),
  #                      na.value="gray70")+
  #scale_x_continuous(expand=c(0,0),breaks=c(331,seq(335,530,by=5)))+
  labs(x="",y="")+theme_classic(base_size=9)+
  coord_equal()+theme(axis.text.x=element_text(angle=90,hjust=1,vjust=0.6,face="bold",size=10),axis.text.y=element_text(face="bold",size=10))+
  facet_wrap(~target,nrow=6)+
  guides(y.sec=guide_axis_label_trans())+
  geom_text(aes(label=wildtype_indicator),size=2,color="gray10")+
  theme(strip.text.x = element_text(size = 18))

p1
invisible(dev.print(pdf, paste(config$final_variant_scores_dir,"/lib40_heatmap_SSM_delta-normAUC-by-target_tri.pdf",sep="")))
```

Next, SARS2mono sera

```{r heatmap_DMS_AUC-by-target_SARS2mono-sera, fig.width=25,fig.height=15,fig.align="center", dpi=500,dev="png",echo=T}
p1 <- ggplot(temp[measurement=="SARS2mono_AUC",],aes(site_SARS2,mutant))+geom_tile(aes(fill=value),color="black",lwd=0.1)+
  scale_fill_gradientn(colours=c("#FFFFFF","#003366"),limits=c(0,8.5),na.value="gray70")+
  #scale_fill_gradientn(colours=c("#FFFFFF","#FFFFFF","#003366"),limits=c(5,12),values=c(0,1/7,7/7),na.value="yellow")+ #three notches in case I want to 'censor' closer to the 5 boundary condition
  #scale_x_continuous(expand=c(0,0),breaks=c(331,seq(335,530,by=5)))+
  labs(x="",y="")+theme_classic(base_size=9)+
  coord_equal()+theme(axis.text.x=element_text(angle=90,hjust=1,vjust=0.6,face="bold",size=10),axis.text.y=element_text(face="bold",size=10))+
  facet_wrap(~target,nrow=6)+
  guides(y.sec=guide_axis_label_trans())+
  geom_text(aes(label=wildtype_indicator),size=2,color="gray10")+
  theme(strip.text.x = element_text(size = 18))

p1
invisible(dev.print(pdf, paste(config$final_variant_scores_dir,"/lib40_heatmap_SSM_AUC-by-target_SARS2mono.pdf",sep="")))
```

Same but illustrating delta_AUC by target (each delta noramlized to its wt background), for SARS2mono sera.

```{r heatmap_DMS_delta-AUC-by-target_SARS2mono-sera, fig.width=25,fig.height=15,fig.align="center", dpi=500,dev="png",echo=T}
p1 <- ggplot(temp[measurement=="delta_SARS2mono_AUC",],aes(site_SARS2,mutant))+geom_tile(aes(fill=value),color="black",lwd=0.1)+
  scale_fill_gradientn(colours=c("#A94E35","#A94E35","#F48365","#FFFFFF","#7378B9","#383C6C","#383C6C"),
                       limits=c(-8,2),
                       values=c(0/10,4/10,6/10,8/10,8.5/10,9/10, 10/10),
                       na.value="gray70")+
  #scale_x_continuous(expand=c(0,0),breaks=c(331,seq(335,530,by=5)))+
  labs(x="",y="")+theme_classic(base_size=9)+
  coord_equal()+theme(axis.text.x=element_text(angle=90,hjust=1,vjust=0.6,face="bold",size=10),axis.text.y=element_text(face="bold",size=10))+
  facet_wrap(~target,nrow=6)+
  guides(y.sec=guide_axis_label_trans())+
  geom_text(aes(label=wildtype_indicator),size=2,color="gray10")+
  theme(strip.text.x = element_text(size = 18))

p1
invisible(dev.print(pdf, paste(config$final_variant_scores_dir,"/lib40_heatmap_SSM_delta-AUC-by-target_SARS2mono.pdf",sep="")))
```

expression-normalized/censored, SARS2mono

```{r heatmap_DMS_delta-normAUC-by-target_SARS2mono-sera, fig.width=25,fig.height=15,fig.align="center", dpi=500,dev="png",echo=T}
p1 <- ggplot(temp[measurement=="delta_SARS2mono_normAUC",],
             aes(site_SARS2,mutant))+geom_tile(aes(fill=value),color="black",lwd=0.1)+
  scale_fill_gradientn(colours=c("#A94E35","#A94E35","#F48365","#FFFFFF","#7378B9","#383C6C"), #if showing blue color
                       limits=c(-5,2),
                       values=c(0/7,3/7,4/7,5/7,6/7,7/7),
                       na.value="gray70")+
  # scale_fill_gradientn(colours=c("#A94E35","#A94E35","#F48365","#FFFFFF","#FFFFFF","#FFFFFF","#FFFFFF"), #if censoring >0 blue
  #                      limits=c(-5,2),
  #                      values=c(0/7,3/7,4/7,5/7,5.5/7,6/7, 7/7),
  #                      na.value="gray70")+
  #scale_x_continuous(expand=c(0,0),breaks=c(331,seq(335,530,by=5)))+
  labs(x="",y="")+theme_classic(base_size=9)+
  coord_equal()+theme(axis.text.x=element_text(angle=90,hjust=1,vjust=0.6,face="bold",size=10),axis.text.y=element_text(face="bold",size=10))+
  facet_wrap(~target,nrow=6)+
  guides(y.sec=guide_axis_label_trans())+
  geom_text(aes(label=wildtype_indicator),size=2,color="gray10")+
  theme(strip.text.x = element_text(size = 18))

p1
invisible(dev.print(pdf, paste(config$final_variant_scores_dir,"/lib40_heatmap_SSM_delta-normAUC-by-target_SARS2mono.pdf",sep="")))
```

Last, RmYN02mono sera

```{r heatmap_DMS_AUC-by-target_RmYN02mono-sera, fig.width=25,fig.height=15,fig.align="center", dpi=500,dev="png",echo=T}
p1 <- ggplot(temp[measurement=="RmYN02mono_AUC",],aes(site_SARS2,mutant))+geom_tile(aes(fill=value),color="black",lwd=0.1)+
  scale_fill_gradientn(colours=c("#FFFFFF","#003366"),limits=c(0,8.5),na.value="gray70")+
  #scale_fill_gradientn(colours=c("#FFFFFF","#FFFFFF","#003366"),limits=c(5,12),values=c(0,1/7,7/7),na.value="yellow")+ #three notches in case I want to 'censor' closer to the 5 boundary condition
  #scale_x_continuous(expand=c(0,0),breaks=c(331,seq(335,530,by=5)))+
  labs(x="",y="")+theme_classic(base_size=9)+
  coord_equal()+theme(axis.text.x=element_text(angle=90,hjust=1,vjust=0.6,face="bold",size=10),axis.text.y=element_text(face="bold",size=10))+
  facet_wrap(~target,nrow=6)+
  guides(y.sec=guide_axis_label_trans())+
  geom_text(aes(label=wildtype_indicator),size=2,color="gray10")+
  theme(strip.text.x = element_text(size = 18))

p1
invisible(dev.print(pdf, paste(config$final_variant_scores_dir,"/lib40_heatmap_SSM_AUC-by-target_RmYN02mono.pdf",sep="")))
```

Same but illustrating delta_AUC by target (each delta noramlized to its wt background), for RmYN02mono sera.

```{r heatmap_DMS_delta-AUC-by-target_RmYN02mono-sera, fig.width=25,fig.height=15,fig.align="center", dpi=500,dev="png",echo=T}
p1 <- ggplot(temp[measurement=="delta_RmYN02mono_AUC",],aes(site_SARS2,mutant))+geom_tile(aes(fill=value),color="black",lwd=0.1)+
  scale_fill_gradientn(colours=c("#A94E35","#A94E35","#F48365","#FFFFFF","#7378B9","#383C6C","#383C6C"),
                       limits=c(-8,2),
                       values=c(0/10,4/10,6/10,8/10,8.5/10,9/10, 10/10),
                       na.value="gray70")+
  #scale_x_continuous(expand=c(0,0),breaks=c(331,seq(335,530,by=5)))+
  labs(x="",y="")+theme_classic(base_size=9)+
  coord_equal()+theme(axis.text.x=element_text(angle=90,hjust=1,vjust=0.6,face="bold",size=10),axis.text.y=element_text(face="bold",size=10))+
  facet_wrap(~target,nrow=6)+
  guides(y.sec=guide_axis_label_trans())+
  geom_text(aes(label=wildtype_indicator),size=2,color="gray10")+
  theme(strip.text.x = element_text(size = 18))

p1
invisible(dev.print(pdf, paste(config$final_variant_scores_dir,"/lib40_heatmap_SSM_delta-AUC-by-target_RmYN02mono.pdf",sep="")))
```

expression-normalized/censored, RmYN02mono

```{r heatmap_DMS_delta-normAUC-by-target_RmYN02mono-sera, fig.width=25,fig.height=15,fig.align="center", dpi=500,dev="png",echo=T}
p1 <- ggplot(temp[measurement=="delta_RmYN02mono_normAUC",],
             aes(site_SARS2,mutant))+geom_tile(aes(fill=value),color="black",lwd=0.1)+
  scale_fill_gradientn(colours=c("#A94E35","#A94E35","#F48365","#FFFFFF","#7378B9","#383C6C"), #if showing blue color
                       limits=c(-5,2),
                       values=c(0/7,3/7,4/7,5/7,6/7,7/7),
                       na.value="gray70")+
  # scale_fill_gradientn(colours=c("#A94E35","#A94E35","#F48365","#FFFFFF","#FFFFFF","#FFFFFF","#FFFFFF"), #if censoring >0 blue
  #                      limits=c(-5,2),
  #                      values=c(0/7,3/7,4/7,5/7,5.5/7,6/7, 7/7),
  #                      na.value="gray70")+
  #scale_x_continuous(expand=c(0,0),breaks=c(331,seq(335,530,by=5)))+
  labs(x="",y="")+theme_classic(base_size=9)+
  coord_equal()+theme(axis.text.x=element_text(angle=90,hjust=1,vjust=0.6,face="bold",size=10),axis.text.y=element_text(face="bold",size=10))+
  facet_wrap(~target,nrow=6)+
  guides(y.sec=guide_axis_label_trans())+
  geom_text(aes(label=wildtype_indicator),size=2,color="gray10")+
  theme(strip.text.x = element_text(size = 18))

p1
invisible(dev.print(pdf, paste(config$final_variant_scores_dir,"/lib40_heatmap_SSM_delta-normAUC-by-target_RmYN02mono.pdf",sep="")))
```

Let's also output heatmaps by background, comparing the four vaccine formulations.

For SARS-CoV-2_WH1:

```{r heatmap_DMS_AUC-by-sera_SARS-CoV-2_WH1, fig.width=25,fig.height=10,fig.align="center", dpi=500,dev="png",echo=T}
p1 <- ggplot(temp[target=="SARS-CoV-2_WH1" & measurement %in% c("tetra_AUC","tri_AUC","SARS2mono_AUC","RmYN02mono_AUC"),], 
             aes(site_SARS2,mutant))+geom_tile(aes(fill=value),color="black",lwd=0.1)+
  scale_fill_gradientn(colours=c("#FFFFFF","#003366"),limits=c(0,8.5),na.value="gray70")+
  #scale_fill_gradientn(colours=c("#FFFFFF","#FFFFFF","#003366"),limits=c(5,12),values=c(0,1/7,7/7),na.value="yellow")+ #three notches in case I want to 'censor' closer to the 5 boundary condition
  #scale_x_continuous(expand=c(0,0),breaks=c(331,seq(335,530,by=5)))+
  labs(x="",y="")+theme_classic(base_size=9)+
  coord_equal()+theme(axis.text.x=element_text(angle=90,hjust=1,vjust=0.6,face="bold",size=10),axis.text.y=element_text(face="bold",size=10))+
  facet_wrap(~measurement,nrow=4)+
  guides(y.sec=guide_axis_label_trans())+
  geom_text(aes(label=wildtype_indicator),size=2,color="gray10")+
  theme(strip.text.x = element_text(size = 18))

p1
invisible(dev.print(pdf, paste(config$final_variant_scores_dir,"/lib40_heatmap_SSM_AUC-by-sera_SARS-CoV-2_WH1.pdf",sep="")))
```

Same but illustrating delta_AUC by sera (each delta noramlized to its wt background), for SARS-CoV-2_WH1

```{r heatmap_DMS_delta-AUC-by-sera_SARS-CoV-2_WH1, fig.width=25,fig.height=10,fig.align="center", dpi=500,dev="png",echo=T}
p1 <- ggplot(temp[target=="SARS-CoV-2_WH1" & measurement %in% c("delta_tetra_AUC","delta_tri_AUC","delta_SARS2mono_AUC","delta_RmYN02mono_AUC"),], 
             aes(site_SARS2,mutant))+geom_tile(aes(fill=value),color="black",lwd=0.1)+
  scale_fill_gradientn(colours=c("#A94E35","#A94E35","#F48365","#FFFFFF","#7378B9","#383C6C","#383C6C"),
                       limits=c(-8,2),
                       values=c(0/10,4/10,6/10,8/10,8.5/10,9/10, 10/10),
                       na.value="gray70")+
  #scale_x_continuous(expand=c(0,0),breaks=c(331,seq(335,530,by=5)))+
  labs(x="",y="")+theme_classic(base_size=9)+
  coord_equal()+theme(axis.text.x=element_text(angle=90,hjust=1,vjust=0.6,face="bold",size=10),axis.text.y=element_text(face="bold",size=10))+
  facet_wrap(~measurement,nrow=4)+
  guides(y.sec=guide_axis_label_trans())+
  geom_text(aes(label=wildtype_indicator),size=2,color="gray10")+
  theme(strip.text.x = element_text(size = 18))

p1
invisible(dev.print(pdf, paste(config$final_variant_scores_dir,"/lib40_heatmap_SSM_delta-AUC-by-sera_SARS-CoV-2_WH1.pdf",sep="")))
```

Same but illustrating delta_normAUC by sera (expresion-norm/cens, each delta noramlized to its wt background), for SARS-CoV-2_WH1

```{r heatmap_DMS_delta-normAUC-by-sera_SARS-CoV-2_WH1, fig.width=25,fig.height=10,fig.align="center", dpi=500,dev="png",echo=T}
p1 <- ggplot(temp[target=="SARS-CoV-2_WH1" & measurement %in% c("delta_tetra_normAUC","delta_tri_normAUC","delta_SARS2mono_normAUC","delta_RmYN02mono_normAUC"),], 
             aes(site_SARS2,mutant))+geom_tile(aes(fill=value),color="black",lwd=0.1)+
  scale_fill_gradientn(colours=c("#A94E35","#A94E35","#F48365","#FFFFFF","#7378B9","#383C6C"), #if showing blue color
                       limits=c(-5,2),
                       values=c(0/7,3/7,4/7,5/7,6/7,7/7),
                       na.value="gray70")+
  # scale_fill_gradientn(colours=c("#A94E35","#A94E35","#F48365","#FFFFFF","#FFFFFF","#FFFFFF","#FFFFFF"), #if censoring >0 blue
  #                      limits=c(-5,2),
  #                      values=c(0/7,3/7,4/7,5/7,5.5/7,6/7, 7/7),
  #                      na.value="gray70")+
  #scale_x_continuous(expand=c(0,0),breaks=c(331,seq(335,530,by=5)))+
  labs(x="",y="")+theme_classic(base_size=9)+
  coord_equal()+theme(axis.text.x=element_text(angle=90,hjust=1,vjust=0.6,face="bold",size=10),axis.text.y=element_text(face="bold",size=10))+
  facet_wrap(~measurement,nrow=4)+
  guides(y.sec=guide_axis_label_trans())+
  geom_text(aes(label=wildtype_indicator),size=2,color="gray10")+
  theme(strip.text.x = element_text(size = 18))

p1
invisible(dev.print(pdf, paste(config$final_variant_scores_dir,"/lib40_heatmap_SSM_delta-normAUC-by-sera_SARS-CoV-2_WH1.pdf",sep="")))
```


For RshSTT182:

```{r heatmap_DMS_AUC-by-sera_RshSTT182, fig.width=25,fig.height=10,fig.align="center", dpi=500,dev="png",echo=T}
p1 <- ggplot(temp[target=="RshSTT182" & measurement %in% c("tetra_AUC","tri_AUC","SARS2mono_AUC","RmYN02mono_AUC"),], 
             aes(site_SARS2,mutant))+geom_tile(aes(fill=value),color="black",lwd=0.1)+
  scale_fill_gradientn(colours=c("#FFFFFF","#003366"),limits=c(0,8.5),na.value="gray70")+
  #scale_fill_gradientn(colours=c("#FFFFFF","#FFFFFF","#003366"),limits=c(5,12),values=c(0,1/7,7/7),na.value="yellow")+ #three notches in case I want to 'censor' closer to the 5 boundary condition
  #scale_x_continuous(expand=c(0,0),breaks=c(331,seq(335,530,by=5)))+
  labs(x="",y="")+theme_classic(base_size=9)+
  coord_equal()+theme(axis.text.x=element_text(angle=90,hjust=1,vjust=0.6,face="bold",size=10),axis.text.y=element_text(face="bold",size=10))+
  facet_wrap(~measurement,nrow=4)+
  guides(y.sec=guide_axis_label_trans())+
  geom_text(aes(label=wildtype_indicator),size=2,color="gray10")+
  theme(strip.text.x = element_text(size = 18))

p1
invisible(dev.print(pdf, paste(config$final_variant_scores_dir,"/lib40_heatmap_SSM_AUC-by-sera_RshSTT182.pdf",sep="")))
```

Same but illustrating delta_AUC by sera (each delta noramlized to its wt background), for RshSTT182

```{r heatmap_DMS_delta-AUC-by-sera_RshSTT182, fig.width=25,fig.height=10,fig.align="center", dpi=500,dev="png",echo=T}
p1 <- ggplot(temp[target=="RshSTT182" & measurement %in% c("delta_tetra_AUC","delta_tri_AUC","delta_SARS2mono_AUC","delta_RmYN02mono_AUC"),], 
             aes(site_SARS2,mutant))+geom_tile(aes(fill=value),color="black",lwd=0.1)+
  scale_fill_gradientn(colours=c("#A94E35","#A94E35","#F48365","#FFFFFF","#7378B9","#383C6C","#383C6C"),
                       limits=c(-8,2),
                       values=c(0/10,4/10,6/10,8/10,8.5/10,9/10, 10/10),
                       na.value="gray70")+
  #scale_x_continuous(expand=c(0,0),breaks=c(331,seq(335,530,by=5)))+
  labs(x="",y="")+theme_classic(base_size=9)+
  coord_equal()+theme(axis.text.x=element_text(angle=90,hjust=1,vjust=0.6,face="bold",size=10),axis.text.y=element_text(face="bold",size=10))+
  facet_wrap(~measurement,nrow=4)+
  guides(y.sec=guide_axis_label_trans())+
  geom_text(aes(label=wildtype_indicator),size=2,color="gray10")+
  theme(strip.text.x = element_text(size = 18))

p1
invisible(dev.print(pdf, paste(config$final_variant_scores_dir,"/lib40_heatmap_SSM_delta-AUC-by-sera_RshSTT182.pdf",sep="")))
```

Same but illustrating delta_normAUC by sera (expresion-norm/cens, each delta noramlized to its wt background), for RshSTT182

```{r heatmap_DMS_delta-normAUC-by-sera_RshSTT182, fig.width=25,fig.height=10,fig.align="center", dpi=500,dev="png",echo=T}
p1 <- ggplot(temp[target=="RshSTT182" & measurement %in% c("delta_tetra_normAUC","delta_tri_normAUC","delta_SARS2mono_normAUC","delta_RmYN02mono_normAUC"),], 
             aes(site_SARS2,mutant))+geom_tile(aes(fill=value),color="black",lwd=0.1)+
  scale_fill_gradientn(colours=c("#A94E35","#A94E35","#F48365","#FFFFFF","#7378B9","#383C6C"), #if showing blue color
                       limits=c(-5,2),
                       values=c(0/7,3/7,4/7,5/7,6/7,7/7),
                       na.value="gray70")+
  # scale_fill_gradientn(colours=c("#A94E35","#A94E35","#F48365","#FFFFFF","#FFFFFF","#FFFFFF","#FFFFFF"), #if censoring >0 blue
  #                      limits=c(-5,2),
  #                      values=c(0/7,3/7,4/7,5/7,5.5/7,6/7, 7/7),
  #                      na.value="gray70")+
  #scale_x_continuous(expand=c(0,0),breaks=c(331,seq(335,530,by=5)))+
  labs(x="",y="")+theme_classic(base_size=9)+
  coord_equal()+theme(axis.text.x=element_text(angle=90,hjust=1,vjust=0.6,face="bold",size=10),axis.text.y=element_text(face="bold",size=10))+
  facet_wrap(~measurement,nrow=4)+
  guides(y.sec=guide_axis_label_trans())+
  geom_text(aes(label=wildtype_indicator),size=2,color="gray10")+
  theme(strip.text.x = element_text(size = 18))

p1
invisible(dev.print(pdf, paste(config$final_variant_scores_dir,"/lib40_heatmap_SSM_delta-normAUC-by-sera_RshSTT182.pdf",sep="")))
```

For SARS-CoV-1_Urbani:

```{r heatmap_DMS_AUC-by-sera_SARS-CoV-1_Urbani, fig.width=25,fig.height=10,fig.align="center", dpi=500,dev="png",echo=T}
p1 <- ggplot(temp[target=="SARS-CoV-1_Urbani" & measurement %in% c("tetra_AUC","tri_AUC","SARS2mono_AUC","RmYN02mono_AUC"),], 
             aes(site_SARS2,mutant))+geom_tile(aes(fill=value),color="black",lwd=0.1)+
  scale_fill_gradientn(colours=c("#FFFFFF","#003366"),limits=c(0,8.5),na.value="gray70")+
  #scale_fill_gradientn(colours=c("#FFFFFF","#FFFFFF","#003366"),limits=c(5,12),values=c(0,1/7,7/7),na.value="yellow")+ #three notches in case I want to 'censor' closer to the 5 boundary condition
  #scale_x_continuous(expand=c(0,0),breaks=c(331,seq(335,530,by=5)))+
  labs(x="",y="")+theme_classic(base_size=9)+
  coord_equal()+theme(axis.text.x=element_text(angle=90,hjust=1,vjust=0.6,face="bold",size=10),axis.text.y=element_text(face="bold",size=10))+
  facet_wrap(~measurement,nrow=4)+
  guides(y.sec=guide_axis_label_trans())+
  geom_text(aes(label=wildtype_indicator),size=2,color="gray10")+
  theme(strip.text.x = element_text(size = 18))

p1
invisible(dev.print(pdf, paste(config$final_variant_scores_dir,"/lib40_heatmap_SSM_AUC-by-sera_SARS-CoV-1_Urbani.pdf",sep="")))
```

Same but illustrating delta_AUC by sera (each delta noramlized to its wt background), for SARS-CoV-1_Urbani

```{r heatmap_DMS_delta-AUC-by-sera_SARS-CoV-1_Urbani, fig.width=25,fig.height=10,fig.align="center", dpi=500,dev="png",echo=T}
p1 <- ggplot(temp[target=="SARS-CoV-1_Urbani" & measurement %in% c("delta_tetra_AUC","delta_tri_AUC","delta_SARS2mono_AUC","delta_RmYN02mono_AUC"),], 
             aes(site_SARS2,mutant))+geom_tile(aes(fill=value),color="black",lwd=0.1)+
  scale_fill_gradientn(colours=c("#A94E35","#A94E35","#F48365","#FFFFFF","#7378B9","#383C6C","#383C6C"),
                       limits=c(-8,2),
                       values=c(0/10,4/10,6/10,8/10,8.5/10,9/10, 10/10),
                       na.value="gray70")+
  #scale_x_continuous(expand=c(0,0),breaks=c(331,seq(335,530,by=5)))+
  labs(x="",y="")+theme_classic(base_size=9)+
  coord_equal()+theme(axis.text.x=element_text(angle=90,hjust=1,vjust=0.6,face="bold",size=10),axis.text.y=element_text(face="bold",size=10))+
  facet_wrap(~measurement,nrow=4)+
  guides(y.sec=guide_axis_label_trans())+
  geom_text(aes(label=wildtype_indicator),size=2,color="gray10")+
  theme(strip.text.x = element_text(size = 18))

p1
invisible(dev.print(pdf, paste(config$final_variant_scores_dir,"/lib40_heatmap_SSM_delta-AUC-by-sera_SARS-CoV-1_Urbani.pdf",sep="")))
```

Same but illustrating delta_normAUC by sera (expresion-norm/cens, each delta noramlized to its wt background), for SARS-CoV-1_Urbani

```{r heatmap_DMS_delta-normAUC-by-sera_SARS-CoV-1_Urbani, fig.width=25,fig.height=10,fig.align="center", dpi=500,dev="png",echo=T}
p1 <- ggplot(temp[target=="SARS-CoV-1_Urbani" & measurement %in% c("delta_tetra_normAUC","delta_tri_normAUC","delta_SARS2mono_normAUC","delta_RmYN02mono_normAUC"),], 
             aes(site_SARS2,mutant))+geom_tile(aes(fill=value),color="black",lwd=0.1)+
  scale_fill_gradientn(colours=c("#A94E35","#A94E35","#F48365","#FFFFFF","#7378B9","#383C6C"), #if showing blue color
                       limits=c(-5,2),
                       values=c(0/7,3/7,4/7,5/7,6/7,7/7),
                       na.value="gray70")+
  # scale_fill_gradientn(colours=c("#A94E35","#A94E35","#F48365","#FFFFFF","#FFFFFF","#FFFFFF","#FFFFFF"), #if censoring >0 blue
  #                      limits=c(-5,2),
  #                      values=c(0/7,3/7,4/7,5/7,5.5/7,6/7, 7/7),
  #                      na.value="gray70")+
  #scale_x_continuous(expand=c(0,0),breaks=c(331,seq(335,530,by=5)))+
  labs(x="",y="")+theme_classic(base_size=9)+
  coord_equal()+theme(axis.text.x=element_text(angle=90,hjust=1,vjust=0.6,face="bold",size=10),axis.text.y=element_text(face="bold",size=10))+
  facet_wrap(~measurement,nrow=4)+
  guides(y.sec=guide_axis_label_trans())+
  geom_text(aes(label=wildtype_indicator),size=2,color="gray10")+
  theme(strip.text.x = element_text(size = 18))

p1
invisible(dev.print(pdf, paste(config$final_variant_scores_dir,"/lib40_heatmap_SSM_delta-normAUC-by-sera_SARS-CoV-1_Urbani.pdf",sep="")))
```

For RsYN04:

```{r heatmap_DMS_AUC-by-sera_RsYN04, fig.width=25,fig.height=10,fig.align="center", dpi=500,dev="png",echo=T}
p1 <- ggplot(temp[target=="RsYN04" & measurement %in% c("tetra_AUC","tri_AUC","SARS2mono_AUC","RmYN02mono_AUC"),], 
             aes(site_SARS2,mutant))+geom_tile(aes(fill=value),color="black",lwd=0.1)+
  scale_fill_gradientn(colours=c("#FFFFFF","#003366"),limits=c(0,8.5),na.value="gray70")+
  #scale_fill_gradientn(colours=c("#FFFFFF","#FFFFFF","#003366"),limits=c(5,12),values=c(0,1/7,7/7),na.value="yellow")+ #three notches in case I want to 'censor' closer to the 5 boundary condition
  #scale_x_continuous(expand=c(0,0),breaks=c(331,seq(335,530,by=5)))+
  labs(x="",y="")+theme_classic(base_size=9)+
  coord_equal()+theme(axis.text.x=element_text(angle=90,hjust=1,vjust=0.6,face="bold",size=10),axis.text.y=element_text(face="bold",size=10))+
  facet_wrap(~measurement,nrow=4)+
  guides(y.sec=guide_axis_label_trans())+
  geom_text(aes(label=wildtype_indicator),size=2,color="gray10")+
  theme(strip.text.x = element_text(size = 18))

p1
invisible(dev.print(pdf, paste(config$final_variant_scores_dir,"/lib40_heatmap_SSM_AUC-by-sera_RsYN04.pdf",sep="")))
```

Same but illustrating delta_AUC by sera (each delta noramlized to its wt background), for RsYN04

```{r heatmap_DMS_delta-AUC-by-sera_RsYN04, fig.width=25,fig.height=10,fig.align="center", dpi=500,dev="png",echo=T}
p1 <- ggplot(temp[target=="RsYN04" & measurement %in% c("delta_tetra_AUC","delta_tri_AUC","delta_SARS2mono_AUC","delta_RmYN02mono_AUC"),], 
             aes(site_SARS2,mutant))+geom_tile(aes(fill=value),color="black",lwd=0.1)+
  scale_fill_gradientn(colours=c("#A94E35","#A94E35","#F48365","#FFFFFF","#7378B9","#383C6C","#383C6C"),
                       limits=c(-8,2),
                       values=c(0/10,4/10,6/10,8/10,8.5/10,9/10, 10/10),
                       na.value="gray70")+
  #scale_x_continuous(expand=c(0,0),breaks=c(331,seq(335,530,by=5)))+
  labs(x="",y="")+theme_classic(base_size=9)+
  coord_equal()+theme(axis.text.x=element_text(angle=90,hjust=1,vjust=0.6,face="bold",size=10),axis.text.y=element_text(face="bold",size=10))+
  facet_wrap(~measurement,nrow=4)+
  guides(y.sec=guide_axis_label_trans())+
  geom_text(aes(label=wildtype_indicator),size=2,color="gray10")+
  theme(strip.text.x = element_text(size = 18))

p1
invisible(dev.print(pdf, paste(config$final_variant_scores_dir,"/lib40_heatmap_SSM_delta-AUC-by-sera_RsYN04.pdf",sep="")))
```

Same but illustrating delta_normAUC by sera (expresion-norm/cens, each delta noramlized to its wt background), for RsYN04

```{r heatmap_DMS_delta-normAUC-by-sera_RsYN04, fig.width=25,fig.height=10,fig.align="center", dpi=500,dev="png",echo=T}
p1 <- ggplot(temp[target=="RsYN04" & measurement %in% c("delta_tetra_normAUC","delta_tri_normAUC","delta_SARS2mono_normAUC","delta_RmYN02mono_normAUC"),], 
             aes(site_SARS2,mutant))+geom_tile(aes(fill=value),color="black",lwd=0.1)+
  scale_fill_gradientn(colours=c("#A94E35","#A94E35","#F48365","#FFFFFF","#7378B9","#383C6C"), #if showing blue color
                       limits=c(-5,2),
                       values=c(0/7,3/7,4/7,5/7,6/7,7/7),
                       na.value="gray70")+
  # scale_fill_gradientn(colours=c("#A94E35","#A94E35","#F48365","#FFFFFF","#FFFFFF","#FFFFFF","#FFFFFF"), #if censoring >0 blue
  #                      limits=c(-5,2),
  #                      values=c(0/7,3/7,4/7,5/7,5.5/7,6/7, 7/7),
  #                      na.value="gray70")+
  #scale_x_continuous(expand=c(0,0),breaks=c(331,seq(335,530,by=5)))+
  labs(x="",y="")+theme_classic(base_size=9)+
  coord_equal()+theme(axis.text.x=element_text(angle=90,hjust=1,vjust=0.6,face="bold",size=10),axis.text.y=element_text(face="bold",size=10))+
  facet_wrap(~measurement,nrow=4)+
  guides(y.sec=guide_axis_label_trans())+
  geom_text(aes(label=wildtype_indicator),size=2,color="gray10")+
  theme(strip.text.x = element_text(size = 18))

p1
invisible(dev.print(pdf, paste(config$final_variant_scores_dir,"/lib40_heatmap_SSM_delta-normAUC-by-sera_RsYN04.pdf",sep="")))
```
For PRD-0038:

```{r heatmap_DMS_AUC-by-sera_PRD-0038, fig.width=25,fig.height=10,fig.align="center", dpi=500,dev="png",echo=T}
p1 <- ggplot(temp[target=="PRD-0038" & measurement %in% c("tetra_AUC","tri_AUC","SARS2mono_AUC","RmYN02mono_AUC"),], 
             aes(site_SARS2,mutant))+geom_tile(aes(fill=value),color="black",lwd=0.1)+
  scale_fill_gradientn(colours=c("#FFFFFF","#003366"),limits=c(0,8.5),na.value="gray70")+
  #scale_fill_gradientn(colours=c("#FFFFFF","#FFFFFF","#003366"),limits=c(5,12),values=c(0,1/7,7/7),na.value="yellow")+ #three notches in case I want to 'censor' closer to the 5 boundary condition
  #scale_x_continuous(expand=c(0,0),breaks=c(331,seq(335,530,by=5)))+
  labs(x="",y="")+theme_classic(base_size=9)+
  coord_equal()+theme(axis.text.x=element_text(angle=90,hjust=1,vjust=0.6,face="bold",size=10),axis.text.y=element_text(face="bold",size=10))+
  facet_wrap(~measurement,nrow=4)+
  guides(y.sec=guide_axis_label_trans())+
  geom_text(aes(label=wildtype_indicator),size=2,color="gray10")+
  theme(strip.text.x = element_text(size = 18))

p1
invisible(dev.print(pdf, paste(config$final_variant_scores_dir,"/lib40_heatmap_SSM_AUC-by-sera_PRD-0038.pdf",sep="")))
```

Same but illustrating delta_AUC by sera (each delta noramlized to its wt background), for PRD-0038

```{r heatmap_DMS_delta-AUC-by-sera_PRD-0038, fig.width=25,fig.height=10,fig.align="center", dpi=500,dev="png",echo=T}
p1 <- ggplot(temp[target=="PRD-0038" & measurement %in% c("delta_tetra_AUC","delta_tri_AUC","delta_SARS2mono_AUC","delta_RmYN02mono_AUC"),], 
             aes(site_SARS2,mutant))+geom_tile(aes(fill=value),color="black",lwd=0.1)+
  scale_fill_gradientn(colours=c("#A94E35","#A94E35","#F48365","#FFFFFF","#7378B9","#383C6C","#383C6C"),
                       limits=c(-8,2),
                       values=c(0/10,4/10,6/10,8/10,8.5/10,9/10, 10/10),
                       na.value="gray70")+
  #scale_x_continuous(expand=c(0,0),breaks=c(331,seq(335,530,by=5)))+
  labs(x="",y="")+theme_classic(base_size=9)+
  coord_equal()+theme(axis.text.x=element_text(angle=90,hjust=1,vjust=0.6,face="bold",size=10),axis.text.y=element_text(face="bold",size=10))+
  facet_wrap(~measurement,nrow=4)+
  guides(y.sec=guide_axis_label_trans())+
  geom_text(aes(label=wildtype_indicator),size=2,color="gray10")+
  theme(strip.text.x = element_text(size = 18))

p1
invisible(dev.print(pdf, paste(config$final_variant_scores_dir,"/lib40_heatmap_SSM_delta-AUC-by-sera_PRD-0038.pdf",sep="")))
```

Same but illustrating delta_normAUC by sera (expresion-norm/cens, each delta noramlized to its wt background), for PRD-0038

```{r heatmap_DMS_delta-normAUC-by-sera_PRD-0038, fig.width=25,fig.height=10,fig.align="center", dpi=500,dev="png",echo=T}
p1 <- ggplot(temp[target=="PRD-0038" & measurement %in% c("delta_tetra_normAUC","delta_tri_normAUC","delta_SARS2mono_normAUC","delta_RmYN02mono_normAUC"),], 
             aes(site_SARS2,mutant))+geom_tile(aes(fill=value),color="black",lwd=0.1)+
  scale_fill_gradientn(colours=c("#A94E35","#A94E35","#F48365","#FFFFFF","#7378B9","#383C6C"), #if showing blue color
                       limits=c(-5,2),
                       values=c(0/7,3/7,4/7,5/7,6/7,7/7),
                       na.value="gray70")+
  # scale_fill_gradientn(colours=c("#A94E35","#A94E35","#F48365","#FFFFFF","#FFFFFF","#FFFFFF","#FFFFFF"), #if censoring >0 blue
  #                      limits=c(-5,2),
  #                      values=c(0/7,3/7,4/7,5/7,5.5/7,6/7, 7/7),
  #                      na.value="gray70")+
  #scale_x_continuous(expand=c(0,0),breaks=c(331,seq(335,530,by=5)))+
  labs(x="",y="")+theme_classic(base_size=9)+
  coord_equal()+theme(axis.text.x=element_text(angle=90,hjust=1,vjust=0.6,face="bold",size=10),axis.text.y=element_text(face="bold",size=10))+
  facet_wrap(~measurement,nrow=4)+
  guides(y.sec=guide_axis_label_trans())+
  geom_text(aes(label=wildtype_indicator),size=2,color="gray10")+
  theme(strip.text.x = element_text(size = 18))

p1
invisible(dev.print(pdf, paste(config$final_variant_scores_dir,"/lib40_heatmap_SSM_delta-normAUC-by-sera_PRD-0038.pdf",sep="")))
```
For RmYN02:

```{r heatmap_DMS_AUC-by-sera_RmYN02, fig.width=25,fig.height=10,fig.align="center", dpi=500,dev="png",echo=T}
p1 <- ggplot(temp[target=="RmYN02" & measurement %in% c("tetra_AUC","tri_AUC","SARS2mono_AUC","RmYN02mono_AUC"),], 
             aes(site_SARS2,mutant))+geom_tile(aes(fill=value),color="black",lwd=0.1)+
  scale_fill_gradientn(colours=c("#FFFFFF","#003366"),limits=c(0,8.5),na.value="gray70")+
  #scale_fill_gradientn(colours=c("#FFFFFF","#FFFFFF","#003366"),limits=c(5,12),values=c(0,1/7,7/7),na.value="yellow")+ #three notches in case I want to 'censor' closer to the 5 boundary condition
  #scale_x_continuous(expand=c(0,0),breaks=c(331,seq(335,530,by=5)))+
  labs(x="",y="")+theme_classic(base_size=9)+
  coord_equal()+theme(axis.text.x=element_text(angle=90,hjust=1,vjust=0.6,face="bold",size=10),axis.text.y=element_text(face="bold",size=10))+
  facet_wrap(~measurement,nrow=4)+
  guides(y.sec=guide_axis_label_trans())+
  geom_text(aes(label=wildtype_indicator),size=2,color="gray10")+
  theme(strip.text.x = element_text(size = 18))

p1
invisible(dev.print(pdf, paste(config$final_variant_scores_dir,"/lib40_heatmap_SSM_AUC-by-sera_RmYN02.pdf",sep="")))
```

Same but illustrating delta_AUC by sera (each delta noramlized to its wt background), for RmYN02

```{r heatmap_DMS_delta-AUC-by-sera_RmYN02, fig.width=25,fig.height=10,fig.align="center", dpi=500,dev="png",echo=T}
p1 <- ggplot(temp[target=="RmYN02" & measurement %in% c("delta_tetra_AUC","delta_tri_AUC","delta_SARS2mono_AUC","delta_RmYN02mono_AUC"),], 
             aes(site_SARS2,mutant))+geom_tile(aes(fill=value),color="black",lwd=0.1)+
  scale_fill_gradientn(colours=c("#A94E35","#A94E35","#F48365","#FFFFFF","#7378B9","#383C6C","#383C6C"),
                       limits=c(-8,2),
                       values=c(0/10,4/10,6/10,8/10,8.5/10,9/10, 10/10),
                       na.value="gray70")+
  #scale_x_continuous(expand=c(0,0),breaks=c(331,seq(335,530,by=5)))+
  labs(x="",y="")+theme_classic(base_size=9)+
  coord_equal()+theme(axis.text.x=element_text(angle=90,hjust=1,vjust=0.6,face="bold",size=10),axis.text.y=element_text(face="bold",size=10))+
  facet_wrap(~measurement,nrow=4)+
  guides(y.sec=guide_axis_label_trans())+
  geom_text(aes(label=wildtype_indicator),size=2,color="gray10")+
  theme(strip.text.x = element_text(size = 18))

p1
invisible(dev.print(pdf, paste(config$final_variant_scores_dir,"/lib40_heatmap_SSM_delta-AUC-by-sera_RmYN02.pdf",sep="")))
```

Same but illustrating delta_normAUC by sera (expresion-norm/cens, each delta noramlized to its wt background), for RmYN02

```{r heatmap_DMS_delta-normAUC-by-sera_RmYN02, fig.width=25,fig.height=10,fig.align="center", dpi=500,dev="png",echo=T}
p1 <- ggplot(temp[target=="RmYN02" & measurement %in% c("delta_tetra_normAUC","delta_tri_normAUC","delta_SARS2mono_normAUC","delta_RmYN02mono_normAUC"),], 
             aes(site_SARS2,mutant))+geom_tile(aes(fill=value),color="black",lwd=0.1)+
  scale_fill_gradientn(colours=c("#A94E35","#A94E35","#F48365","#FFFFFF","#7378B9","#383C6C"), #if showing blue color
                       limits=c(-5,2),
                       values=c(0/7,3/7,4/7,5/7,6/7,7/7),
                       na.value="gray70")+
  # scale_fill_gradientn(colours=c("#A94E35","#A94E35","#F48365","#FFFFFF","#FFFFFF","#FFFFFF","#FFFFFF"), #if censoring >0 blue
  #                      limits=c(-5,2),
  #                      values=c(0/7,3/7,4/7,5/7,5.5/7,6/7, 7/7),
  #                      na.value="gray70")+
  #scale_x_continuous(expand=c(0,0),breaks=c(331,seq(335,530,by=5)))+
  labs(x="",y="")+theme_classic(base_size=9)+
  coord_equal()+theme(axis.text.x=element_text(angle=90,hjust=1,vjust=0.6,face="bold",size=10),axis.text.y=element_text(face="bold",size=10))+
  facet_wrap(~measurement,nrow=4)+
  guides(y.sec=guide_axis_label_trans())+
  geom_text(aes(label=wildtype_indicator),size=2,color="gray10")+
  theme(strip.text.x = element_text(size = 18))

p1
invisible(dev.print(pdf, paste(config$final_variant_scores_dir,"/lib40_heatmap_SSM_delta-normAUC-by-sera_RmYN02.pdf",sep="")))
```

Save output files.

```{r outputs}
dt_mutant[,.(target,wildtype,site,site_SARS2,mutant,mutation,mutation_SARS2,
            tetra_AUC,sd_tetra_AUC,delta_tetra_AUC,delta_tetra_normAUC,n_bc_tetra_AUC,
            tri_AUC,sd_tri_AUC,delta_tri_AUC,delta_tri_normAUC,n_bc_tri_AUC,
            SARS2mono_AUC,sd_SARS2mono_AUC,delta_SARS2mono_AUC,delta_SARS2mono_normAUC,n_bc_SARS2mono_AUC,
            RmYN02mono_AUC,sd_RmYN02mono_AUC,delta_RmYN02mono_AUC,delta_RmYN02mono_normAUC,n_bc_RmYN02mono_AUC,
            expr,delta_expr)] %>%
  mutate_if(is.numeric, round, digits=5) %>%
  write.csv(file=config$final_variant_scores_lib40_file, row.names=F,quote=F)
```





